botocore/credentials.py

```diff
@@ -1064,19 +1076,18 @@ def _load_creds_via_assume_role(self, profile_name):
             mfa_prompter=self._prompter,
             cache=self.cache,
         )
-        base_credentials = fetcher.fetch_credentials()
-
+        refresher = fetcher.fetch_credentials
         if mfa_serial is not None:
-            refresher = create_mfa_serial_refresher()
-        else:
-            refresher = fetcher.fetch_credentials
+            refresher = create_mfa_serial_refresher(refresher)
 
-        expiration = _parse_if_needed(base_credentials['expiry_time'])
+        # The initial credentials are empty and the expiration time is set
+        # to now so that we can delay the call to assume role until it is
+        # strictly needed.
         return RefreshableCredentials(
-            access_key=base_credentials['access_key'],
-            secret_key=base_credentials['secret_key'],
-            token=base_credentials['token'],
-            expiry_time=expiration,
+            access_key='',
```

Comment:
Taked to you offline about this as well. But I would prefer we have a explicit value as the access and secret key (i.e. `NOT_SET`, `NEEDS_REFRESH`) to indicate on the first load of these we need to do the refresh logic.